### PR TITLE
Test/6490 ubuntu support

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Some improvements in ubuntu core features and documentation

### CODE REVIEW

File: testsuite/docs/optional.md [hunk 1]
```diff
@@ -92,6 +92,38 @@ Inside of the testsuite, the scenarios that are tagged with
 are executed only if the CentOS minion is available.
 
 
+### Testing with a Ubuntu minion
+
+Using a Ubuntu minion with the testsuite is not mandatory.
+
+If you do not want a CentOS minion, do not define `UBUNTUMINION` environment
```
Comment (amannch) [inline testsuite/docs/optional.md]: "CentOS minion" in the text should be replaced with "Ubuntu minion"

File: testsuite/features/core_ubuntu_salt.feature [hunk 2]
```diff
@@ -1,48 +1,55 @@
...
+    And I should see a "Bootstrap Minions" text
+    And I enter the hostname of "ubuntu-minion" as "hostname"
     And I enter "22" as "port"
     And I enter "root" as "user"
     And I enter "linux" as "password"
     And I select "1-UBUNTU-TEST" from "activationKeys"
-    And I select the hostname of the proxy from "proxies"
-    And I click on "Bootstrap"
-    And I wait until I see "Successfully bootstrapped host! " text
+    When I click on "Bootstrap"
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: And

File: testsuite/features/core_ubuntu_salt.feature [hunk 3]
```diff
@@ -1,48 +1,55 @@
...
+    Then I wait until I see "Successfully bootstrapped host!" text
     And I navigate to "rhn/systems/Overview.do" page
     And I wait until I see the name of "ubuntu-minion", refreshing the page
     And I wait until onboarding is completed for "ubuntu-minion"
+    And I query latest Salt changes on ubuntu system "ubuntu-minion"
 
-@proxy
 @ubuntu_minion
-  Scenario: Check connection from Ubuntu minion to proxy
+  Scenario: Schedule an OpenSCAP audit job for the Ubuntu minion
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: This should not be in a core feature.

I suggest you commit "as is", and let @lkotek move it to a secondary feature for you. He just has to do as he did for CentOS.

Alternatively, take your inspiration from @lkotek's work, and split this yourself.

File: testsuite/features/core_ubuntu_salt.feature [hunk 4]
```diff
@@ -57,16 +64,16 @@ Feature: Be able to bootstrap an Ubuntu minion and do some basic operations on i
 
 @ubuntu_minion
   Scenario: Run a remote command on the Ubuntu minion
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: Same, this kind of tests is now done in the secondary features (we want core to be leightweight).

File: testsuite/features/core_ubuntu_salt_ssh.feature [hunk 5]
```diff
@@ -7,21 +7,6 @@
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt_ssh.feature]: Bump date to 2019 in the copyright.

We want only *one* ubuntu feature in core. So this should be renamed to min_ubuntu_salt_ssh.feature, assuming that by default ubuntu is "salt" and not "salt ssh".
->
same, i suggest you commit "as is" and let @lkotek move it to non-core.

File: testsuite/features/core_ubuntu_salt.feature [hunk 6]
```diff
@@ -1,48 +1,55 @@
 # Copyright (c) 2017-2018 SUSE LLC
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: Please use 2017-2019 in the copyright

Also, rename to `core_min_ubuntu_salt.feature please` (name conventions are in the documentation).

File: testsuite/features/core_ubuntu_salt.feature [hunk 7]
```diff
@@ -57,16 +64,16 @@ Feature: Be able to bootstrap an Ubuntu minion and do some basic operations on i
 
 @ubuntu_minion
   Scenario: Run a remote command on the Ubuntu minion
-    Given I am authorized as "testing" with password "testing"
-    When I follow "Salt"
+    Given I am authorized
+    And I follow "Salt"
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: When

File: testsuite/features/core_ubuntu_salt.feature [hunk 8]
```diff
@@ -57,16 +64,16 @@ Feature: Be able to bootstrap an Ubuntu minion and do some basic operations on i
...
     And I follow "Remote Commands"
-    Then I should see a "Remote Commands" text
-    When I enter command "cat /etc/os-release"
+    And I should see a "Remote Commands" text
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: Then

File: testsuite/features/core_ubuntu_salt.feature [hunk 9]
```diff
@@ -57,16 +64,16 @@ Feature: Be able to bootstrap an Ubuntu minion and do some basic operations on i
...
+    And I enter command "cat /etc/os-release"
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: When

File: testsuite/features/core_ubuntu_salt.feature [hunk 10]
```diff
@@ -57,16 +64,16 @@ Feature: Be able to bootstrap an Ubuntu minion and do some basic operations on i
...
     And I enter target "*ubuntu*"
     And I click on preview
-    And I click on run
+    When I click on run
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: And

File: testsuite/features/core_ubuntu_salt.feature [hunk 11]
```diff
@@ -57,16 +64,16 @@ Feature: Be able to bootstrap an Ubuntu minion and do some basic operations on i
...
     Then I should see "ubuntu-minion" hostname
-    When I wait for "15" seconds
+    And I wait until I see "show response" text
```
Comment (Bischoff) [inline testsuite/features/core_ubuntu_salt.feature]: When

File: testsuite/run_sets/testsuite.yml [hunk 12]
```diff
@@ -22,18 +22,18 @@
   # one of: core_proxy_register_as_trad_with_script.feature
   #         core_proxy_register_as_minion_with_script.feature
   #         core_proxy_register_as_minion_with_gui.feature
-- features/core_proxy_register_as_minion_with_gui.feature
-- features/core_proxy_branch_network.feature
+#- features/core_proxy_register_as_minion_with_gui.feature
```
Comment (Bischoff) [inline testsuite/run_sets/testsuite.yml]: No. This looks like an improper merge conflict resolution. Please revert as it was before.

File: testsuite/run_sets/testsuite.yml [hunk 13]
```diff
@@ -22,18 +22,18 @@
...
 # initialize clients
-- features/core_trad_register_client.feature
-- features/core_min_bootstrap.feature
-- features/core_min_salt_ssh.feature
-- features/core_centos_tradclient.feature
-- features/core_centos_salt.feature
-- features/min_centos_salt_install_package_and_patch.feature
-- features/core_centos_salt_ssh.feature
-#- features/core_ubuntu_salt.feature
-#- features/core_ubuntu_salt_ssh.feature
```
Comment (Bischoff) [inline testsuite/run_sets/testsuite.yml]: only one ubuntu feature in "core" - see with @lkotek

File: testsuite/features/core_ubuntu_salt.feature (hunk 4 above)
Comment (srbarrios) [reply testsuite/features/core_ubuntu_salt.feature]: Totally agree. But we will do in a separate PR.

File: testsuite/features/core_ubuntu_salt.feature [hunk 14]
```diff
@@ -1,48 +1,55 @@
...
+#    When I install Salt packages from "ubuntu-minion"
+#    And I start salt-minion on "ubuntu-minion"
+
 @ubuntu_minion
   Scenario: Bootstrap an Ubuntu minion
     Given I am authorized
-    When I go to the bootstrapping page
-    Then I should see a "Bootstrap Minions" text
-    When I enter the hostname of "ubuntu-minion" as "hostname"
+    And I go to the bootstrapping page
```
[the remark this answers is not in the document]
Comment (srbarrios) [reply testsuite/features/core_ubuntu_salt.feature]: You know that we will never agree on that...
I think each scenario must follow the structure Given And And And... When And And And ... Then And And And..
But it's a battle I don't want to still fighting. Gonna change as you are suggesting

File: testsuite/features/core_ubuntu_salt.feature (hunk 3 above)
Comment (srbarrios) [reply testsuite/features/core_ubuntu_salt.feature]: Totally agree, we will address that in a separate PR.

File: testsuite/run_sets/testsuite.yml (hunk 13 above)
Comment (srbarrios) [reply testsuite/run_sets/testsuite.yml]: Let's address it another PR, so it's easier to merge that one in ubuntu-support branch.
Remember that PR is not going to Manager-3.2, but to ubuntu-support branch (dev branch)

File: testsuite/run_sets/testsuite.yml (hunk 12 above)
Comment (srbarrios) [reply testsuite/run_sets/testsuite.yml]: No. I commented it in ubuntu-support branch.
We don't need proxy, and that will make pass our tests faster, until we have a stable testsuite, then we enable again all these features ;)

**Comment by srbarrios** [comment]
> @srbarrios I think you need to rebase your branch on top of latest `ubuntu-support`, picking only the commits you intend to add. I can see some commits in the PR which are already added to the `ubuntu-support` branch before.

As I pull and merge the upstream branch ubuntu-support several times, I think doing the rebase of all the commits it's even worse. So if you don't have a better idea, I think I will create a new branch clean and apply a patch of all the changes I maded.

**Comment by cbbayburt** [comment]
> > @srbarrios I think you need to rebase your branch on top of latest `ubuntu-support`, picking only the commits you intend to add. I can see some commits in the PR which are already added to the `ubuntu-support` branch before.
> 
> As I pull and merge the upstream branch ubuntu-support several times, I think doing the rebase of all the commits it's even worse. So if you don't have a better idea, I think I will create a new branch clean and apply a patch of all the changes I maded.

You can either use `rebase -i` and drop unwanted commits in the process, or you can create a new branch and a PR and cherry-pick your commits there. The best choice is whichever you feel more familiar to.

**Comment by srbarrios** [comment]
> > > @srbarrios I think you need to rebase your branch on top of latest `ubuntu-support`, picking only the commits you intend to add. I can see some commits in the PR which are already added to the `ubuntu-support` branch before.
> > 
> > 
> > As I pull and merge the upstream branch ubuntu-support several times, I think doing the rebase of all the commits it's even worse. So if you don't have a better idea, I think I will create a new branch clean and apply a patch of all the changes I maded.
> 
> You can either use `rebase -i` and drop unwanted commits in the process, or you can create a new branch and a PR and cherry-pick your commits there. The best choice is whichever you feel more familiar to.

Ah yeah, I did squash instead of drop ^^ my bad. Now is too late. Going to cherry-pick...

**Comment by srbarrios** [comment]
Sorry guys.. closing that one and open a cleaner one here: https://github.com/uyuni-project/uyuni/pull/511

**Comment by Bischoff** [comment]
@srbarrios Yeah, sometimes it's better to start anew :-) .

**Pull request closed without merging**